src/components/icons/create-styled-icon.js

```diff
@@ -33,55 +33,48 @@ const getSizeStyle = size => {
   }
 };
 
-const getColor = theme => {
-  if (!theme) return 'inherit';
-  switch (theme) {
-    case 'black':
-      return vars.colorSolid;
-    case 'grey':
-      return vars.colorNeutral60;
-    case 'white':
-      return vars.colorSurface;
-    case 'blue':
-      return vars.colorInfo;
-    case 'green':
-      return vars.colorPrimary;
-    case 'green-light':
-      return vars.colorPrimary40;
-    case 'orange':
-      return vars.colorWarning;
-    case 'red':
-      return vars.colorError;
-    default: {
-      invariant(
-        theme,
-        `ui-kit/Icon: the specified theme '${theme}' is not supported.`
-      );
-      return 'inherit';
-    }
+const capitalize = s => s[0].toUpperCase() + s.slice(1);
+
+const getColor = (color, theme) => {
+  if (!color) return 'inherit';
+  const overwrittenVars = {
+    ...vars,
+    ...theme,
+  };
+
+  const iconColor = overwrittenVars[`color${capitalize(color)}`];
```

Comment:
is this more like what you were thinking @tdeekens ?

Reply:
Ah yeah. Thanks!